configs/components/curl.rb

```diff
@@ -31,9 +31,10 @@
   configure_options << "--with-ssl=#{settings[:prefix]}"
 
   extra_cflags = []
-  if platform.is_cross_compiled? && platform.is_macos?
+  if platform.architecture == 'arm64' && platform.is_macos?
     extra_cflags << '-mmacosx-version-min=11.0 -arch arm64' if platform.name =~ /osx-11/
     extra_cflags << '-mmacosx-version-min=12.0 -arch arm64' if platform.name =~ /osx-12/
+    extra_cflags << '-mmacosx-version-min=13.0 -arch arm64' if platform.name =~ /osx-13/
```

Comment:
Do we actually need to specify the minimum version when doing native macOS 13 ARM compiles? If so, then it would be nice to not have to update this for each macOS version (and just keep the code how it was)?

However, if the minimum version is required, then can we drop the `arch`?

```suggestion
if platform.is_macos?
  if platform.is_cross_compiled?
    extra_cflags << '-mmacosx-version-min=11.0 -arch arm64' if platform.name =~ /osx-11/
    extra_cflags << '-mmacosx-version-min=12.0 -arch arm64' if platform.name =~ /osx-12/
  else
    extra_cflags << '-mmacosx-version-min=13.0' if platform.name =~ /osx-13/
  end
end
```